src/index.rs

```diff
@@ -669,15 +643,24 @@ mod tests {
     use std::fs::File;
     #[cfg(target_os = "linux")]
     use std::fs::Permissions;
+    use std::io::Write;
     #[cfg(target_os = "linux")]
     use std::os::unix::fs::PermissionsExt;
```

Comment:
This gives us permissions to write files on linux systems only so some unit tests are failing on mac.

We can use `#[cfg(target_family = "unix")]` instead

Reply:
For context: https://github.com/ARK-Builders/arklib/pull/72#discussion_r1478470500